lib/src/server.dart

```diff
@@ -56,25 +56,54 @@ abstract class Service {
 /// Listens for incoming gRPC calls, dispatching them to a [ServerHandler].
 class Server {
   final Map<String, Service> _services = {};
-  ServerSocket _server;
+  final int port;
+  final SecurityContext _securityContext;
 
+  ServerSocket _insecureServer;
+  SecureServerSocket _secureServer;
   final _connections = <ServerTransportConnection>[];
 
-  final int _port;
+  Server._(List<Service> services, this.port, this._securityContext) {
+    for (final service in services) {
+      _services[service.$name] = service;
+    }
+  }
 
-  Server({int port = 8080}) : _port = port;
+  factory Server.insecure(List<Service> services, {int port}) {
+    return new Server._(services, port ?? 80, null);
+  }
 
-  void addService(Service service) {
-    _services[service.$name] = service;
+  factory Server.secure(List<Service> services,
+      {List<int> certificate,
+      String certificatePassword,
+      List<int> privateKey,
+      String privateKeyPassword,
+      int port}) {
+    final context = createSecurityContext(true);
+    if (privateKey != null) {
```

Comment:
Again: what happens if passwords are null, but certificate and/or private key are not? Do we fail in the right way, if encryption is used?